Content.Server/PAI/PAISystem.cs

```diff
@@ -1,70 +1,138 @@
-        {
-            if (component.LastUser == null)
-                return;
+        // Ownership tag
+        var val = Loc.GetString("pai-system-pai-name", ("owner", component.LastUser));
 
-            // Ownership tag
-            var val = Loc.GetString("pai-system-pai-name", ("owner", component.LastUser));
+        // TODO Identity? People shouldn't dox-themselves by carrying around a PAI.
+        // But having the pda's name permanently be "old lady's PAI" is weird.
+        // Changing the PAI's identity in a way that ties it to the owner's identity also seems weird.
+        // Cause then you could remotely figure out information about the owner's equipped items.
 
-            // TODO Identity? People shouldn't dox-themselves by carrying around a PAI.
-            // But having the pda's name permanently be "old lady's PAI" is weird.
-            // Changing the PAI's identity in a way that ties it to the owner's identity also seems weird.
-            // Cause then you could remotely figure out information about the owner's equipped items.
+        _metaData.SetEntityName(uid, val);
+    }
 
-            _metaData.SetEntityName(uid, val);
-        }
+    private void OnMindRemoved(EntityUid uid, PAIComponent component, MindRemovedMessage args)
+    {
+        // Mind was removed, shutdown the PAI.
+        PAITurningOff(uid);
+    }
 
-        private void OnMindRemoved(EntityUid uid, PAIComponent component, MindRemovedMessage args)
+    private void OnMicrowaved(EntityUid uid, PAIComponent comp, BeingMicrowavedEvent args)
+    {
+        // name will always be scrambled whether it gets bricked or not, this is the reward
+        ScrambleName(uid, comp);
+
+        // randomly brick it
+        if (_random.Prob(comp.BrickChance))
+        {
+            _popup.PopupEntity(Loc.GetString(comp.BrickPopup), uid, PopupType.LargeCaution);
+            _toggleableGhostRole.Wipe(uid);
+            RemComp<PAIComponent>(uid);
+            RemComp<ToggleableGhostRoleComponent>(uid);
+        }
+        else
         {
-            // Mind was removed, shutdown the PAI.
-            PAITurningOff(uid);
+            // you are lucky...
+            _popup.PopupEntity(Loc.GetString(comp.ScramblePopup), uid, PopupType.Large);
         }
+    }
 
-        public void PAITurningOff(EntityUid uid)
+    private void ScrambleName(EntityUid uid, PAIComponent comp)
+    {
+        // randomly replace random characters from the old name
+        var oldName = Name(uid);
+        var name = new StringBuilder(oldName.Length);
+        var named = false;
+        foreach (var character in oldName)
         {
-            //  Close the instrument interface if it was open
-            //  before closing
-            if (HasComp<ActiveInstrumentComponent>(uid) && TryComp<ActorComponent>(uid, out var actor))
+            // only scramble the owner name, don't scramble "'s pAI"
+            if (character == '\'')
```

Comment:
if its purely random the original name length and some of the characters are lost
currently you have incentive to try microwave multiple times since itll get progressively more unrecognizable

Reply:
I think it's fine if you just have to microwave it once. Your current approach looks like it would break if a name contains `'`, or if pai-system-pai-name is changed in a localization.

Reply:
correct and correct guess ill change it